Object/Manager.php

```diff
@@ -75,20 +75,30 @@ public function search(Criteria $criteria)
         /** @var \Doctrine\Common\Persistence\Mapping\ClassMetadata $metadata */
         $metadata = $this->getManager()->getClassMetadata($this->getClass());
 
+        $identifierFieldsNames = $metadata->getIdentifier();
+        $identifierFieldName = reset($identifierFieldsNames);
+        $countQueryBuilder = $this->getManager()->createQueryBuilder();
+        $expressionBuilder = $countQueryBuilder->expr();
+        $whereConditions = $expressionBuilder->eq(1, 1);
         foreach ($criteria as $key => $value) {
             if (!$metadata->hasField($key) && !$metadata->hasAssociation($key)) {
                 $criteria->remove($key);
             }
+            $whereConditions = $expressionBuilder->andX($whereConditions, $expressionBuilder->eq("e.{$key}", $value));
         }
 
-        $allObjects = $this->getRepository()->findBy(
-            $criteria->toArray(),
-            $criteria->getOrderBy()
+        $filteringCriteria = $criteria->toArray();
+        $objects = $this->getRepository()->findBy(
+            $filteringCriteria,
+            /*$orderBy =*/$criteria->getOrderBy(),
```

Comment:
This is the main query, There is another query for the count, But this is the main one with the criteria, order, limit and offset